pkg/patch/patch_test.go

```diff
@@ -291,3 +298,228 @@ func TestGetOSVersion(t *testing.T) {
 		})
 	}
 }
+
+func TestGeneratePatchedTag(t *testing.T) {
+	testCases := []struct {
+		name                 string
+		dockerImageName      string
+		userSuppliedPatchTag string
+		expectedPatchedTag   string
+	}{
+		{
+			name:                 "NoTag_NoUserSupplied",
+			dockerImageName:      "docker.io/library/alpine",
+			userSuppliedPatchTag: "",
+			expectedPatchedTag:   defaultPatchedTagSuffix,
+		},
+		{
+			name:                 "WithTag_NoUserSupplied",
+			dockerImageName:      "docker.io/redhat/ubi9:latest",
+			userSuppliedPatchTag: "",
+			expectedPatchedTag:   "latest-patched",
+		},
+		{
+			name:                 "WithTag_UserSupplied",
+			dockerImageName:      "docker.io/librari/ubuntu:jammy-20231004",
+			userSuppliedPatchTag: "20231004-patched",
+			expectedPatchedTag:   "20231004-patched",
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			named, _ := reference.ParseNormalizedNamed(tc.dockerImageName)
+			patchedTag := generatePatchedTag(named, tc.userSuppliedPatchTag)
+			if patchedTag != tc.expectedPatchedTag {
+				t.Errorf("expected patchedTag to be %s but got %s", tc.expectedPatchedTag, patchedTag)
+			}
+		})
+	}
+}
+
+func TestUpdateManifest(t *testing.T) {
```

Comment:
In order to understand this test I need a definition of what an "errpackage" is. Could be done via a comment or a changed variable name.

Also, I'm struggling to understand what unversioned means here, (did some _very quick_ digging around the types associated with the code under test but as I was missing your guidance I could figure it out myself! Also, I am about to head out the door so didn't have the time to pour over the code)

Use this test to explain what is going on here. Don't change the underlying code, just provide some short commentary here on what is happening,

The audience for the commentary is between an end-user and a new contributor, assume the audience knows how to use copa and that this test gives them crucial insight in to how copa behaves.

Reply:
Other than that, the test is good, I can kind of see what is going on. It's just the signage that needs improving here.